driver/apdu/at_win32.c

```diff
@@ -0,0 +1,348 @@
+#define AT_READ_BUFFER_SIZE 4096
+
+static HANDLE hComm;
+static int logic_channel = 0;
+static char *at_cmd_buffer;
+static char at_read_buffer[AT_READ_BUFFER_SIZE];
+static DWORD at_read_buffer_len = 0;
+
+static int at_expect(char **response, const char *expected)
+{
+    char line[AT_BUFFER_SIZE];
+    DWORD bytes_read;
+    char* found_response_data = NULL;
+    int result = -1;
+
+    if (response)
+        *response = NULL;
+
+    while (1)
+    {
+        char *newline = memchr(at_read_buffer, '\n', at_read_buffer_len);
+        if (!newline) {
+            if (at_read_buffer_len >= sizeof(at_read_buffer)) {
+                 fprintf(stderr, "AT response line too long or buffer full\n");
+                 at_read_buffer_len = 0;
+                 return -1;
+            }
+            if (!ReadFile(hComm, at_read_buffer + at_read_buffer_len, sizeof(at_read_buffer) - at_read_buffer_len, &bytes_read, NULL)) {
+                fprintf(stderr, "ReadFile error: %lu\n", GetLastError());
+                return -1;
+            }
+
+            if (bytes_read == 0) {
+                fprintf(stderr, "AT command timeout\n");
+                return -1;
+            }
+
+            at_read_buffer_len += bytes_read;
+            continue;
+        }
+
+        int line_len = (newline - at_read_buffer);
+        memcpy(line, at_read_buffer, line_len);
+        line[line_len] = '\0';
+
+        memmove(at_read_buffer, newline + 1, at_read_buffer_len - line_len - 1);
+        at_read_buffer_len -= (line_len + 1);
+
+        line[strcspn(line, "\r")] = 0;
+
+        if (strlen(line) == 0) {
+            continue;
+        }
+
+        if (getenv("AT_DEBUG"))
+            printf("AT_DEBUG_RX: %s\n", line);
+
+        if (strcmp(line, "ERROR") == 0)
+        {
+            result = -1;
+            goto end;
+        }
+        else if (strcmp(line, "OK") == 0)
+        {
+            result = 0;
+            goto end;
+        }
+        else if (expected && strncmp(line, expected, strlen(expected)) == 0)
+        {
+            free(found_response_data);
+            found_response_data = strdup(line + strlen(expected));
+        }
+    }
+
+end:
+    if (result == 0) {
+        if (response) {
+            *response = found_response_data;
+        } else {
+            free(found_response_data);
+        }
+    } else {
+        free(found_response_data);
+    }
+    return result;
+}
+
+static int at_write_command(const char *cmd) {
+    DWORD bytes_written;
+    if (getenv("AT_DEBUG"))
+        printf("AT_DEBUG_TX: %s", cmd);
+
+    if (!WriteFile(hComm, cmd, strlen(cmd), &bytes_written, NULL)) {
+        fprintf(stderr, "Failed to write to port, error: %lu\n", GetLastError());
+        return -1;
+    }
+    return 0;
+}
+
+static int apdu_interface_connect(struct euicc_ctx *ctx)
+{
+    const char *device;
+    DCB dcb = {0};
+    COMMTIMEOUTS timeouts = {0};
+
+    logic_channel = 0;
+
+    if (!(device = getenv("AT_DEVICE_WIN32")))
```

Comment:
I think we can reuse environment variable name "AT_DEVICE", since apdu/at and apdu/win32 will never be used together.
And add document for it.